lego/apps/files/views.py

```diff
@@ -46,6 +49,23 @@ def create(self, request, *args, **kwargs):
             status=status.HTTP_201_CREATED,
         )
 
+    @decorators.action(
+        detail=True,
+        methods=["PATCH"],
+        permission_classes=[permissions.IsAuthenticated],
+    )
+    def set_save_for_use(self, request, *args, **kwargs):
+        if request.user.has_perm(EDIT, Event) is False:
+            raise PermissionDenied()
+        file = self.get_object()
+        serializer = FileSaveForUseSerializer(data=request.data)
+        if serializer.is_valid():
+            file.save_for_use = serializer.validated_data["save_for_use"]
+            file.save()
+            return Response(data=serializer.data, status=status.HTTP_200_OK)
```

Comment:
I can't see that you ever check permissions on the image here?

I can elaborate: If you have event EDIT perm, then it looks to me like you can set this variable on any image as long as you have the file key. I.e. you can access an image you should not be able to access as long as you have the key and event EDIT perm. 

You need to implement some sort of perm check here. Even though it's crude, I think using the token auth as in the `FileField` should be adequate, as that token should never be shared.